lib/modules/datasource/postprocess-release.ts

```diff
@@ -0,0 +1,59 @@
+import { logger } from '../../logger';
+import type {
+  LookupUpdateConfig,
+  UpdateResult,
+} from '../../workers/repository/process/lookup/types';
+import { getDatasourceFor } from './common';
+import type { Release } from './types';
+
+type Config = Partial<LookupUpdateConfig & UpdateResult>;
+
+export async function postprocessRelease(
+  config: Config,
+  release: Release,
+): Promise<Release | null> {
+  const { datasource } = config;
+
+  const ds = datasource && getDatasourceFor(datasource);
+  if (!ds) {
+    logger.warn(
+      { datasource },
+      'Failed to resolve datasource during release postprocessing',
+    );
+    return release;
+  }
+
+  if (!ds.postprocessRelease) {
+    return release;
+  }
+
+  const { packageName } = config;
+  if (!packageName) {
+    logger.warn(
+      { datasource },
+      'Release postprocessing is not supported for empty `packageName` field',
+    );
+    return release;
+  }
+
+  const registryUrl = config.registryUrl ?? null;
+
+  try {
+    const result = await ds.postprocessRelease(
+      { packageName, registryUrl },
+      release,
+    );
+
+    if (!result) {
+      logger.debug(
+        { packageName, registryUrl, version: release.version },
+        'Rejected release',
+      );
+    }
```

Comment:
Or should this debug be moved down to the datasource layer where a more meaningful reason could be logged?